pkg/scheduler/objects/application_events.go

```diff
@@ -35,11 +36,11 @@ type applicationEvents struct {
 	limiter     *rate.Limiter
 }
 
-func (evt *applicationEvents) sendAppDoesNotFitEvent(request *AllocationAsk) {
+func (evt *applicationEvents) sendAppDoesNotFitEvent(request *AllocationAsk, headroom *resources.Resource) {
 	if !evt.eventSystem.IsEventTrackingEnabled() || !evt.limiter.Allow() {
 		return
 	}
-	message := fmt.Sprintf("Application %s does not fit into %s queue", request.GetApplicationID(), evt.app.queuePath)
+	message := fmt.Sprintf("Application %s does not fit into %s queue, request resoure %s can't fit headroom %s", request.GetApplicationID(), evt.app.queuePath, request.GetAllocatedResource().String(), headroom.String())
```

Comment:
Minor nit no.1: calling String() is not needed, fmt.Sprintf() can recognize the Stringer interface.
Minor nit no.2: slight rewording: "Application %s does not fit into %s queue (request resoure %s, headroom %s)". The proposed message sounds like two separate problems.

Reply:
Thanks, addressed in latest PR.